### ISSUE TITLE
mix deps.get doesn't show the path to conflicting rebar.config when dependencies diverge

### ISSUE THREAD

**Issue by alco** [elixir-lang/elixir#5523]
I'm testing out Elixir v1.4.0-rc.0.

I have a dependency with two conflicting specifications:

```
$ mix deps.get
Dependencies have diverged:
* meck (git://github.com/eproxus/meck.git)
  the dependency meck in rebar.config is overriding a child dependency:

  > In rebar.config:
    {:meck, ~r/.*/, [env: :prod, git: "git://github.com/eproxus/meck.git", tag: "0.8.2", manager: :rebar3]}

  > In rebar.config:
    {:meck, ~r/.*/, [env: :prod, git: "git://github.com/eproxus/meck", tag: "0.8.2", manager: :rebar3]}

  Ensure they match or specify one of the above in your deps and set "override: true"
```

This error report doesn't show which apps define each respective dependency. I can do a text search in my `deps` directory to find it:

```
exometer_core/rebar.config
14:  {meck, ".*", {git, "git://github.com/eproxus/meck.git", {tag,"0.8.2"}}},

folsom/rebar.config
5:    {meck, ".*", {git, "git://github.com/eproxus/meck", {tag, "0.8.2"}}}
```

but this is not straightforward.

Mix already has the info about the projects that contain those conflicting dep specifications, it should print their names in the error report.

**Comment by maxstr** [comment]
I believe this can be closed.

I see:
```
Dependencies have diverged:
* bear (https://github.com/puzza007/bear.git)
  the dependency bear in mix.exs is overriding a child dependency:

  > In mix.exs:
    {:bear, [env: :prod, git: "https://github.com/puzza007/bear.git", tag: "0.8.2", manager: :rebar3]}

  > In deps/folsom/rebar.config:
    {:bear, "0.8.3", [env: :prod, hex: :bear, optional: false, manager: :rebar3]}
```
now.

**Comment by josevalim** [comment]
@alco please confirm and if not fixed can you please provide a mix.exs file?

**Comment by ericmj** [comment]
This should have been fixed in 1.4.1. https://github.com/elixir-lang/elixir/commit/60f051e05d572c11df72434428c09bc101ec341a#diff-7629669db4b5fb40fe5c9b821de5db70R308

**Comment by alco** [comment]
This looks fixed, good job! Sorry for the delay :)
